### PR TITLE
remove the prep keyword argument from tape

### PR DESCRIPTION
**Pull request by timmysilv**
**Context:**
The `prep` kwarg from tape is no longer needed, and is being removed.

**Description of the Change:**
Remove the `prep` kwarg from tape

**Benefits:**
less code!

**Possible Drawbacks:**
N/A

[sc-47809]

### CODE REVIEW

File: doc/development/deprecations.rst [hunk 1]
```diff
@@ -79,6 +73,12 @@ Pending deprecations
 Completed deprecation cycles
 ----------------------------
 
+* The ``prep`` keyword argument in ``QuantumScript`` has been removed from ``QuantumScript``.
```
Comment (timmysilv) [inline doc/development/deprecations.rst:76]:
```suggestion
* The ``prep`` keyword argument in ``QuantumScript`` has been removed.
```

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -17,6 +17,10 @@
 
 <h3>Breaking changes 💔</h3>
 
+* The `prep` keyword argument has been removed from `QuantumScript` and `QuantumTape`.
+  ``StatePrepBase`` operations should be placed at the beginning of the `ops` list instead.
+  [(#)](https://github.com/PennyLaneAI/pennylane/pull/)
```
Comment (timmysilv) [inline doc/releases/changelog-dev.md:22]:
```suggestion
  [(#4756)](https://github.com/PennyLaneAI/pennylane/pull/4756)
```

File: doc/development/deprecations.rst [hunk 3]
```diff
@@ -114,6 +114,12 @@ Completed deprecation cycles
   - Added in v0.32
   - Behaviour changed in v0.33
 
+* The ``prep`` keyword argument in ``QuantumScript`` has been removed.
```
Comment (albi3ro) [inline doc/development/deprecations.rst:117]: Entry needs to be removed around Line 59.

Comment (timmysilv) [reply doc/development/deprecations.rst:111]: oops, bad merge resolution. ty!

**Comment by mudit2812** [comment]
Minor change, but breaking regardless, so let's hold off on merging it until after the release stuff is done. I don't want to end up in another situation like last time when we removed `qml.active_return` and it broke two plugins and caused the qml builds to fail.

**Review by mudit2812 — APPROVED** [review]
Otherwise looks good!

**Review by albi3ro — APPROVED** [review]

**Pull request merged**